test/e2e/common/node/container_probe.go

```diff
@@ -418,6 +418,34 @@ var _ = SIGDescribe("Probing container", func() {
 			framework.Failf("Pod became ready before startupProbe succeeded")
 		}
 	})
+
+	/*
+		Release: v1.21
+		Testname: Set terminationGracePeriodSeconds for livenessProbe
+		Description: A pod with a long terminationGracePeriod is created with a shorter probe-level terminationGracePeriodSeconds. We confirm the shorter termination period is used.
+	*/
+	ginkgo.It("should override timeoutGracePeriodSeconds when LivenessProbe field is set [NodeAlphaFeature:ProbeTerminationGracePeriod]", func() {
+		pod := e2epod.NewAgnhostPod(f.Namespace.Name, "liveness-override-"+string(uuid.NewUUID()), nil, nil, nil, "/bin/sh", "-c", "sleep 1000")
+		longGracePeriod := int64(500)
+		pod.Spec.TerminationGracePeriodSeconds = &longGracePeriod
+
+		// probe will fail since pod has no http endpoints
+		shortGracePeriod := int64(5)
+		pod.Spec.Containers[0].LivenessProbe = &v1.Probe{
+			Handler: v1.Handler{
+				HTTPGet: &v1.HTTPGetAction{
+					Path: "/healthz",
+					Port: intstr.FromInt(8080),
+				},
+			},
+			InitialDelaySeconds:           10,
+			FailureThreshold:              1,
+			TerminationGracePeriodSeconds: &shortGracePeriod,
+		}
+
+		// 10s delay + 10s period + 5s grace period = 25s < 30s << pod-level timeout 500
+		RunLivenessTest(f, pod, 1, time.Second*30)
```

Comment:
There's no equivalent for StartupProbes so I'll add another test similar to the one above tomorrow.

The node-alpha job seems to be broken and isn't running any tests. :/ I'm not sure if the annotation above is enough, I based the selector on the job config: https://github.com/kubernetes/test-infra/blob/6d5f7e5d77c83aa14d299bbd228bca77bdf0ecb1/config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml#L194

Reply:
https://github.com/kubernetes/kubernetes/issues/98220 is the problem. I will see if I can run the e2es locally using the workaround.

Reply:
For this to work you will need to enable `--runtime-config=api/all=true` on the job... but it will NOT work for Node tests.

Reply:
@knabben I changed the annotation to run these as part of the pull-kubernetes-e2e-gce-alpha-features job which has this set. PR to add it: https://github.com/kubernetes/test-infra/pull/21242